source/synthDrivers/_espeak.py

```diff
@@ -366,16 +366,19 @@ def getVariantDict():
 	# Translators: name of the default espeak varient.
 	variantDict={"none": pgettext("espeakVarient", "none")}
 	for fileName in os.listdir(dir):
-		if os.path.isfile("%s\\%s"%(dir,fileName)):
-			file=codecs.open("%s\\%s"%(dir,fileName))
-			for line in file:
-				if line.startswith('name '):
-					temp=line.split(" ")
-					if len(temp) ==2:
-						name=temp[1].rstrip()
-						break
-				name=None
-			file.close()
+		absFilePath = os.path.join(dir, fileName)
+		if os.path.isfile(absFilePath):
+			# In python 3, open assumes the default system encoding by default.
+			# This fails if Windows' "use Unicode UTF-8 for worldwide language support" option is enabled.
+			# The expected encoding is unknown, therefore use latin-1 to stay as close to Python 2 behavior as possible.
+			with open(absFilePath, 'r', encoding="latin-1") as file:

```

Comment:
I don't understand why we wouldn't just open this as UTF-8? With latin-1, we will encounter an error as soon as someone wants to use Japanese/Chinese/Korean characters in a Variant name.

A [stack overflow answer](https://stackoverflow.com/questions/33837551/what-is-the-default-encoding-for-python-2-7-8) points out that `codecs.open` (in Python 2) defaults to ASCII, so if we want the closest behaviour we should use ASCII decoding.

Reply:
The problem is that opening it as utf-8 is the case that fails. It raises an UNicodeDecodeError for an invalid start byte.

The assumption that codecs.open uses ASCII is incorrect, at least on python 2. I tested this by putting an € sign in a file and opening that file with codecs.open on Python 2. It returns python 2 byte strings when reading, so it just returns the raw bytes from the file without applying any decoding.

Note that @michaelDCurran also suggested latin-1 on slack somewhere.

Reply:
Ok fair enough. An alternative is to treat the file as binary, process it as bytes, and only decode the b"name" field. Is there any guidance on the format of these files from the espeak project? How would one name a voice with Japanese characters?

Reply:
All errors are due to the usage of \xa0, non breaking space.
I think I prefer leaving this as is, however I will put filing an issue against espeak-ng on my todo list. I think these variant info files really should be utf-8 and non breaking spaces should be removed from them.